### DISCUSSION TITLE
Known bugs

### DISCUSSION THREAD

**Discussion by eltikia** [eltikia/Perl-1.0#17] [Q&A] [answered]
Are there any known bugs or limitations?

**Comment by Stagyrite** [comment, marked as answer]
Yes, there are, and here's an example. Let's assume you check how much memory _'postgres'_ uses. This can be done by executing:
`echo "q" | top -b | grep "postgres" | awk -F " " '{ memory += $10 } END { print memory }`
Let's assume the numbers were: 0.2, 3.6, 3.5, 0.1, 0.0, 0.0, 3.7, 1.0, 0.6, 0.4, 2.4, 2.5, 2.3, 2.5, 0.4, and 0.4. The output number is 23.6, and it's a valid percentage value. Here's a Perl 1 program generated with the Awk to Perl translator from the 'x2p' subdirectory. It's a slightly modified version of it.
```
#!./perl
eval '$'.$1.'$2;' while $ARGV[0] =~ /^([A-Za-z_]+=)(.*)/ && shift;
                        # process any FOO=bar switches

$, = ' ';               # set output field separator
$\ = "\n";              # set output record separator

while (<>) {
    ($Fld1,$Fld2,$Fld3,$Fld4,$Fld5,$Fld6,$Fld7,$Fld8,$Fld9,$Fld10) = split(' ');
    $memory += $Fld10;
}

print $memory;
```
The output number is now 23.599999999999997868, and that is a slightly different result. The difference is 0.000000000000002132.
